File: Esame_22/JAVA/Studente.java
import java.util.*;
import java.io.*;

public class Studente {

	// Attributi
	private int matricola;
	private String nomeCognomeStudente;
	protected List<Esame> libretto; // memorizza delle coppie fatte da codiceCorso, Voto

	// Costruttore
	public Studente(int matricola, String nomeCognomeStudente) {
		this.matricola = matricola;
		this.nomeCognomeStudente = nomeCognomeStudente;
		this.libretto = new LinkedList<Esame>(); // Assegna a libretto una nuova linked list
													// di Esame che all'inizio sarà vuota
	}

	// Metodi
	public void addEsame(Esame e) {
		libretto.add(e);
	}

	// PUNTO 4: stampare per ogni studente nome, media esami, se 30L allora deve
	// convertirsi in 31
	// e il suo libretto, con codiceCorso e 30 e lode indicato come 30L

	public double getMedia() {
		double somma = 0;
		for (Esame e : libretto)
			somma = somma + e.getVotoAsInt();
		return somma / libretto.size();
	}

	public String toString() {
		return this.nomeCognomeStudente + "\t" + getMedia() + "\t" + libretto;
	} // in libretto ho la toString definita in Esame fatta così: [ ]

	// PUNTO 5: uso il metodo getter per ritornare la matricola
	public int getMatricola() {
		return this.matricola;
	}

	// ricerco il massimo in una lista
	public Esame votoMax() {
		int max = 0; // ha il voto massimo corrente
		Esame maxe = null; // ha l'esame associato a quel voto
		for (Esame e : libretto) {
			if (e.getVotoAsInt() > max) // se voto convertito (metodo nella classe dell'elenco)è maggiore del massimo
										// corrente
			{
				max = e.getVotoAsInt(); // allora aggiorno il massimo
				maxe = e; // ed aggiorno l'esame associato al voto massimo
			}
		}
		return maxe; // restituisco proprio l'esame con voto massimo
	}

	// metodo getter per la stampa nel punto 5
	public String getNomeCognomeStudente() {
		return this.nomeCognomeStudente;
	}

} // fine classe

File: Esame_22/PYTHON/Studente.py
from Esame import Esame
class Studente:

    # COSTRUTTORE
    def __init__(self, matricola , nomeStudente):
        self._matricola = matricola # _ vuol dire che attributo è privato
        self._nomeStudente = nomeStudente
        # lkl = LinkedList()
        # self._libretto = lkl
        self._libretto = [] # libretto dello studente nella forma di un elenco di corsi
        # lo assegno a lista vuota perchè al momento non ci sono esami

    # METODI
    def getMatricola(self):
        return self._matricola

    def getNome(self):
        return self._nomeStudente

    def addEsame(self , e): # e e' un oggetto della classe Esame
        self._libretto.append( e ) # aggiungo l'esame al libretto usando append

    # PUNTO 4: per arrivare a fare la toString
    def getMedia(self):
        tot = 0
        for e in self._libretto: # scandisco tutti gli esami e nel libretto
            tot = tot + e.getVotoAsInt() # per ogni esame chiamo il metodo getVotoAsInt()
        return ( tot/len( self._libretto ) ) # totale voti / lunghezza elenco

    def librString(self): # questo metodo traduce il libretto in una stringa
        res = "" # il risultato inizialmente e' la stringa vuota
        for e in self._libretto: # scandisco tutti gli esami del libretto
            res = res + e.toString() # aggiungo la conversione dell'esaeme al risultato
        return  res # restituisco il risultato

    def toString(self):
        media = self.getMedia()
        libretto = self.librString()
        return self._nomeStudente + "\t" + str(media) + "\t" + str(libretto)

    # PUNTO 5:
    def votoMax(self):
        maxVote = 0 # inizializzo il voto massimo a zero
        for e in self._libretto: # scandisco gli esami del libretto
            if( e.getVotoAsInt() > maxVote ): # prendo il voto come un intero e lo confronto maxVote
                maxVote = e.getVotoAsInt() # assegno maxVote al voto ottenuto come intero
                maxe = e # assegno e a maxe
        return maxe # ritorno maxe
